pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -168,6 +156,24 @@ func (configgen *ConfigGeneratorImpl) BuildSidecarOutboundHTTPRouteConfig(env mo
 	return out
 }
 
+// buildVirtualHostDomains generates the set of domain matches for a service being accessed from
+// a proxy node
+func buildVirtualHostDomains(service *model.Service, port int, node model.Proxy) []string {
+	domains := []string{service.Hostname, fmt.Sprintf("%s:%d", service.Hostname, port)}
+	if !service.MeshExternal {
+		domains = append(domains, generateAltVirtualHosts(service.Hostname, port, node.Domain)...)
+	}
+	if len(service.Address) > 0 {
+		// add a vhost match for the IP (if its non CIDR)
+		cidr := util.ConvertAddressToCidr(service.Address)
```

Comment:
Ug, I did _not_ see that implemented our own CIDR parsing. I'd have stopped that dead in a PR. We should not do that ourselves. Can you please delete this method in the util and just use [`net.ParseCIDR`](https://golang.org/pkg/net/#ParseCIDR).

Reply:
We should be careful. In validation we allow IP or CIDR for external services. `net.ParseCIDR` won't allow IPs.

Reply:
Ah, good call (IMO it's pretty bad that it doesn't just parse the IP as part of the /32 subnet). This isn't so bad given we're returning an Envoy config object anyway.